lib/with-tarball-stream.js

```diff
@@ -107,7 +107,7 @@ function withTarballStream (spec, opts, streamHandler) {
               // Retry once if we have a cache, to clear up any weird conditions.
               // Don't retry network errors, though -- make-fetch-happen has already
               // taken care of making sure we're all set on that front.
-              if (opts.cache && err.code && !err.code.match(/^E\d{3}$/)) {
+              if (opts.cache && err.code && err.code.match && !err.code.match(/^E\d{3}$/)) {
```

Comment:
This doesn't catch the case where `err.code` is a `number`; I think this might be a more effective check:

```suggestion
              if (opts.cache && err.code && !String(err.code).match(/^E\d{3}$/)) {
```

Reply:
Good point. Done!